components/Forms/TransactionForm.js

```diff
@@ -5,26 +5,60 @@ export default function TransactionForm({
   onAddTransaction,
   formType,
   onCloseModal,
+  savingsTransferSum,
 }) {
   function handleSubmit(event) {
     event.preventDefault();
+
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
-    const transactionData = {
-      ...data,
-      amount: parseInt(data.amount, 10),
-      date: data.date ? formatDate(data.date) : formatDate(new Date()),
-    };
+
     if (formType === "saving transaction") {
-      transactionData.category = "Savings transfer";
-      transactionData.description = "Transfer to savings Account";
-      
-      transactionData.internalGoalAllocation = "no";
-      transactionData.type = "Expense"
+      // Saving Transfer to GoalAccount
+      onAddTransaction({
+        category: "Savings transfer",
+        description: "Transfer to savings",
+        type: "Expense",
+        amount: parseInt(data.amount),

```

Comment:
Specify the radix for parseInt
```suggestion
        amount: parseInt(data.amount, 10),

```